### PR TITLE
Deprecate login option userEmail in favor of loginHint, also use with…

### PR DESCRIPTION
**Pull request by bradvogel**
… Google signin.

Fixes https://github.com/meteor/meteor/issues/2422

### CODE REVIEW

**Comment by bradvogel** [comment]
Tests failed but the look flaky.

**Comment by bradvogel** [comment]
@glasser @stubailo quick look over?

**Comment by bradvogel** [comment]
@glasser @stubailo ping :)

**Comment by stubailo** [comment]
@bradvogel curious why you decided to change the option name - is it to be the same as a the Google API option name?

**Comment by bradvogel** [comment]
@stubailo it was a suggestion from @estark37 here: https://github.com/meteor/meteor/pull/2790#issuecomment-63357825

File: packages/meteor-developer/meteor_developer_client.js [hunk 1]
```diff
@@ -29,8 +29,16 @@ var requestCredential = function (options, credentialRequestCompleteCallback) {
...
 
-  if (options && options.userEmail)
+  /**
+   * @deprecated in 1.3.0
+   */
+  if (options && options.userEmail && !options.loginHint) {
+    options.loginHint = options.userEmail;
+  }
+
+  if (options && options.userEmail) {
```
Comment (zimme) [inline packages/meteor-developer/meteor_developer_client.js]: Shouldn't this be used only if `loginHint` isn't specified as on line 35?

File: packages/meteor-developer/meteor_developer_client.js [hunk 2]
```diff
@@ -29,8 +29,16 @@ var requestCredential = function (options, credentialRequestCompleteCallback) {
...
     loginUrl += '&user_email=' + encodeURIComponent(options.userEmail);
```
Comment (zimme) [inline packages/meteor-developer/meteor_developer_client.js]: or maybe you intended on using `options.loginHint` here?

**Comment by stubailo** [comment]
@zimme thanks for jumping in with some comments!

**Comment by bradvogel** [comment]
Thanks guys. I meant `loginHint` instead `userEmail` - careless mistake. Thanks for the catch.

**Comment by zimme** [comment]
No problem, usually try and help out here and there whenever I feel like I have a minute over when looking around =)

**Comment by stubailo** [comment]
Merged! 24718debfdd409c6c1f73f92c5d4d7d3a8322e55

Thanks for the contribution, @bradvogel!

**Comment by zimme** [comment]
Also maybe document that `loginHint` can be some sort of sub string of the google account id when used with the google accounts system. Saw something about that in the docs @bradvogel linked to.

**Comment by stubailo** [comment]
Good point. @bradvogel could you add something about that in a new PR please?

**Comment by bradvogel** [comment]
https://github.com/meteor/meteor/pull/5370

**Pull request closed without merging**